apps/vs-agent/src/utils/WebVhDidRegistrar.ts

```diff
@@ -0,0 +1,213 @@
+import {
+  AgentContext,
+  DidCreateOptions,
+  DidCreateResult,
+  DidDeactivateOptions,
+  DidDeactivateResult,
+  DidDocumentRole,
+  DidRecord,
+  DidRegistrar,
+  DidRepository,
+  DidUpdateOptions,
+  DidUpdateResult,
+  KeyType,
+  Buffer,
+  DidDocumentService,
+  DidCommV1Service,
+  DidDocument,
+} from '@credo-ts/core'
+import * as crypto from '@stablelib/ed25519'
+import {
+  createDID,
+  DIDLog,
+  multibaseDecode,
+  multibaseEncode,
+  MultibaseEncoding,
+  Signer,
+  updateDID,
+  VerificationMethod,
+  Verifier,
+  WitnessParameter,
+  WitnessProofFileEntry,
+} from 'didwebvh-ts'
+
+import { WebvhDidCryptoExt } from './WebvhDidCryptoExt'
+
+interface WebVhDidCreateOptions extends DidCreateOptions {
+  domain: string
+}
+
+interface WebVhDidUpdateOptions extends DidUpdateOptions {
+  log: DIDLog
+  signer: Signer
+  services?: DidDocumentService[]
+  domain?: string
+  updated?: string
+  updateKeys?: string[]
+  verificationMethods?: VerificationMethod[]
+  controller?: string
+  context?: string | string[] | object | object[]
+  alsoKnownAs?: string[]
+  portable?: boolean
+  nextKeyHashes?: string[]
+  witness?: WitnessParameter | null
+  watchers?: string[] | null
+  verifier?: Verifier
+  authentication?: string[]
+  assertionMethod?: string[]
+  keyAgreement?: string[]
+  witnessProofs?: WitnessProofFileEntry[]
+}
+
+export class WebVhDidRegistrar implements DidRegistrar {
+  supportedMethods: string[] = ['webvh']
+
+  public async update(agentContext: AgentContext, options: WebVhDidUpdateOptions): Promise<DidUpdateResult> {
```

Comment:
This doesn't look so nice. Why I need to define an external signer and verifier instance every time I want to update a DID?

Think about the case where I want to add a Linked VP to an existing WebVH DID: I just want to pass the relevant parameters (e.g. services), but I will not be willing to create something internal to this library/registrar, who should make it automatically.